crates/pixi_utils/src/reqwest.rs

```diff
@@ -111,6 +115,20 @@ pub fn build_reqwest_clients(config: Option<&Config>) -> (Client, ClientWithMidd
 
     client_builder = client_builder.with(GCSMiddleware);
 
+    // todo: same auth storage as above
+    // todo: is from_env correct?
+    debug!("Using s3_config: {:?}", s3_config);
+    let s3_middleware = if let Some(s3_config) = s3_config {
+        S3Middleware::new(s3_config, AuthenticationStorage::from_env().unwrap())
+    } else {
+        S3Middleware::new(
+            config.compute_s3_config(),
+            AuthenticationStorage::from_env().unwrap(),
+        )
+    };
```

Comment:
depends on https://github.com/conda/rattler/pull/1026 and https://github.com/prefix-dev/pixi/pull/2909